### PR TITLE
maven: migrate short-term to reload4j v1.2.18

### PR DESCRIPTION
**Pull request by rohityadavcloud**
This migrate to log4j 1.x fork, reload4j 1.2.18.0 which is drop-in
replacement and addresses some immediate CVE and issues.

This will require some manual test of logging, as well as smoketests.

Fixes: (from https://reload4j.qos.ch/)
Standardize and sanitize the build - fixed in 1.2.18.0
CVE-2021-4104 (JMSAppender) - fixed in 1.2.18.0 by hardening
CVE-2022-23302 (JMSSink) - fixed in 1.2.18.1 by hardening
CVE-2019-17571 (SocketServer) - fixed in 1.2.18.0 by hardening
CVE-2020-9493 and CVE-2022-23307 (Chainsaw) - fixed in 1.2.18.1 by hardening
CVE-2022-23305 (JDBCAppender) - fixed in 1.2.18.2 by hardening the component.
broken MDC in newer JDKs - fixed in 1.2.18.0
XML entity injection attack - fixed in 1.2.18.3 by hardening
CVE-2020-9488 (SMTPAppender) fixed in 1.2.18.3 by hardening

### CODE REVIEW

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
Tested locally, the build works - pl help review, advise on the PR cc @DaanHoogland @borisstoyanov @vladimirpetrov @sureshanaparti @nvazquez @weizhouapache @shwstppr @mlsorensen

**Comment by weizhouapache** [comment]
nice !
looking forward to packaging and smoke test results.

File: framework/managed-context/pom.xml [hunk 1]
```diff
@@ -29,9 +29,9 @@
     </parent>
     <dependencies>
         <dependency>
-            <groupId>log4j</groupId>
-            <artifactId>log4j</artifactId>
-            <version>${cs.log4j.version}</version>
```
Comment (DaanHoogland) [inline framework/managed-context/pom.xml:34]: do we (still) need the version here?

File: plugins/integrations/kubernetes-service/pom.xml [hunk 2]
```diff
@@ -86,9 +86,9 @@
             <version>${cs.guava.version}</version>
         </dependency>
         <dependency>
-            <groupId>log4j</groupId>
-            <artifactId>log4j</artifactId>
-            <version>${cs.log4j.version}</version>
```
Comment (DaanHoogland) [inline plugins/integrations/kubernetes-service/pom.xml:91]: do we (still) need the version here?

**Review by DaanHoogland — APPROVED** [review]
looks good

**Comment by DaanHoogland** [comment]
@blueorangutan test matrix

File: framework/managed-context/pom.xml (hunk 1 above)
Comment (rohityadavcloud) [reply framework/managed-context/pom.xml:34]: Not sure, did n't want to refactor/experiment; so I've left all build config as is just replaced log4j with reload4j.

**Comment by DaanHoogland** [comment]
@blueorangutan test matrix

**Comment by rohityadavcloud** [comment]
@blueorangutan test

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@blueorangutan test

**Review by sureshanaparti — APPROVED** [review]
changes LGTM

**Comment by rohityadavcloud** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by rohityadavcloud** [comment]
@blueorangutan test centos8 xenserver-71

**Comment by rohityadavcloud** [comment]
@blueorangutan help

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test centos8 xenserver-71

**Comment by mlsorensen** [comment]
@rohityadavcloud - just looked at the dependency tree with this change. Do we need to move to slf4j 1.7.35? See https://www.slf4j.org/news.html

> In this release, the "slf4j-log4j12" artifact automatically instructs Maven to use the "slf4j-reload4j" artifact instead. As you might have guessed, the "slf4j-reload4j" binding delegates log processing to the reload4j logging framework.

I'm also seeing esapi pulling in log4j 1.2.17, and contrail plugin pulling in log4j 1.2.16.

**Comment by rohityadavcloud** [comment]
Thanks for the review @mlsorensen and the exclusion fix. I'll kick some tests.
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test matrix

**Comment by rohityadavcloud** [comment]
cc @Pearl1594 are the k8s failures intermittent?
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by Pearl1594** [comment]
> cc @Pearl1594 are the k8s failures intermittent? @blueorangutan package

Based on the logs it seems to be due to :
`unable to drain Kubernetes node on VM : testcluster-FDZXL0-control-17ecbae5fe8'`
I haven't encountered this failure before, however, it may have been a one off occurrence wherein, the worker node may not be in ready state and the control node is being drained, which leaves the k8s with no other nodes to schedule pods on. Let's wait for the current test run to complete. If it fails yet again, I'll investigate it. Thanks

**Pull request merged**